annotated.go

```diff
@@ -1300,7 +1317,11 @@ func (at *asAnnotation) results(ann *annotated) (
 
 		newOutResult := reflect.New(resType).Elem()
 		for i := 1; i < resType.NumField(); i++ {
-			newOutResult.Field(i).Set(getResult(i, results))
+			if newOutResult.Field(i).Kind() == reflect.Func {
+				newOutResult.Field(i).Set(getResult(i, results).Convert(newOutResult.Field(i).Type()))
+			} else {
+				newOutResult.Field(i).Set(getResult(i, results))
+			}
```

Comment:
nit: should we flip these conditions to make the control flow more/appear consistent with the check above?
```
if newOutResult.Field(i).Kind() == reflect.Interface {
    newOutResult.Field(i).Set(getResult(i, results))
} else {
    newOutResult.Field(i).Set(getResult(i, results).Convert(newOutResult.Field(i).Type()))
}
```